### PR TITLE
Fix for #3402. Allows spawning an actor via expression.

### PR DESCRIPTION
**Pull request by chrisjhoare**
This fixes the exception when spawning from an expression using the FSharp api. This did require bumping the FSharp version to 4.5 to get it work though.

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Hard test failure caused by these changes in Akka.FSharp

File: src/core/Akka.FSharp/FsApi.fs [hunk 1]
```diff
@@ -374,14 +374,15 @@ module Linq =
 
     let toExpression<'Actor>(f : System.Linq.Expressions.Expression) = 
             match f with
-            | Lambda(_, Invoke(Call(null, Method "ToFSharpFunc", Ar [| Lambda(_, p) |])))
```
[the remark this answers is not in the document]
Comment (chrisjhoare) [reply src/core/Akka.FSharp/FsApi.fs]: Yeah I think the previous signature of this got changed/removed in the move to netstandard and added back in some form from F# 4.5 onwards - but to be honest its a little beyond me. But this does appear to allow the actor to be spawned now.

**Comment by chrisjhoare** [comment]
The test that's failing doesn't appear to be related to these changes. #3407 added this test (the F# serializer and the cluster pub serializer have the same Identifier - 9). I'm not sure #3404 is fixed as these two still have the same identifier as far as I can see. I could probably fix this in the PR by giving the F# one another number e.g. 99 ? I'm not sure how the id range is determined though?

File: src/core/Akka.FSharp/FsApi.fs [hunk 2]
```diff
@@ -29,7 +29,7 @@ module Serialization =
     type ExprSerializer(system) = 
         inherit Serializer(system)
         let fsp = FsPickler.CreateBinarySerializer()
-        override __.Identifier = 9
+        override __.Identifier = 99
```
Comment (Aaronontheweb) [inline src/core/Akka.FSharp/FsApi.fs]: Good call

**Comment by Aaronontheweb** [comment]
Build stage failed because connection to Github was disrupted when TeamCity was fetching latest patch changes. Going to re-run it.

**Comment by Aaronontheweb** [comment]
Previous test stages failed earlier because I had changed some settings around on the build server in order to support https://github.com/akkadotnet/akka.net/pull/3668 - changed those back so we can run specs under the current build system

**Pull request merged**